### PR TITLE
Feat: search by hashtags / cashtags

### PR DESCRIPTION
**Pull request by mattyg**
Implements searching for cashtags / hashtags. 

Currently it copies the search architecture from the profiles module -- generating links from the first 3 characters of a hashtag to the full hashtag string. We may want to refactor this to sharding the entire tag string in 2-3 character increments. 

Also, it currently searches case-insensitively, but preserves the case of each tag returned (i.e. #HOLO and #holo might both appear as search results).

I think it's beyond the scope of this ticket, but there will need to be some refactoring for broader support for RTL languages (see https://blog.twitter.com/engineering/en_us/a/2012/right-to-left-support-for-twitter-mobile)

### CODE REVIEW

File: ui/src/layouts/MainLayout.vue [hunk 1]
```diff
@@ -137,21 +149,44 @@ const search = (
...
+
     if (inputValue === "" || inputValue.length < 3) {
       options.value = [];
     } else {
       try {
         searching.value = true;
         const profilesMap = await profilesStore.value.searchProfiles(
           inputValue
+        );        
+        const hashtags = await searchHashtags(
```
Comment (jost-s) [inline ui/src/layouts/MainLayout.vue:163]:
```suggestion
        const [profilesMap, hashtags, cashtags] = await Promise.all(profilesStore.value.searchProfiles(inputValue), searchHashtags(inputValue), searchCashtags(inputValue));
```

**Comment by jost-s** [comment]
It looks great to me at first sight, clean and simple. I'll review properly shortly.

**Comment by jost-s** [comment]
> Currently it copies the search architecture from the profiles module -- generating links from the first 3 characters of a hashtag to the full hashtag string. We may want to refactor this to sharding the entire tag string in 2-3 character increments.

Agreed, that'll be good.
 
> Also, it currently searches case-insensitively, but preserves the case of each tag returned (i.e. #HOLO and #holo might both appear as search results).

Agreed as well, same for profiles.

> I think it's beyond the scope of this ticket, but there will need to be some refactoring for broader support for RTL languages (see https://blog.twitter.com/engineering/en_us/a/2012/right-to-left-support-for-twitter-mobile)

Yes, definitely beyond this scope. That'd be cool and there is a test with arabic language, but I think that's a relatively low priority right now.

**Comment by mattyg** [comment]
> > Currently it copies the search architecture from the profiles module -- generating links from the first 3 characters of a hashtag to the full hashtag string. We may want to refactor this to sharding the entire tag string in 2-3 character increments.
> 
> Agreed, that'll be good.

Maybe it makes sense to pull this into a separate ticket, and call it okay for now as a first-pass. I'd like to get a really solid implementation here and in profiles.

**Comment by jost-s** [comment]
> > > Currently it copies the search architecture from the profiles module -- generating links from the first 3 characters of a hashtag to the full hashtag string. We may want to refactor this to sharding the entire tag string in 2-3 character increments.
> > 
> > 
> > Agreed, that'll be good.
> 
> Maybe it makes sense to pull this into a separate ticket, and call it okay for now as a first-pass. I'd like to get a really solid implementation of sharded search here and in profiles.

Absolutely.

**Review by jost-s — APPROVED** [review]
I'm really happy with this. Please resolve conflict and lint.

**Pull request merged**